### ISSUE TITLE
Close in uartdmx is checked for return value... 

### ISSUE THREAD

**Issue by rewolff** [OpenLightingProject/ola#1144]
Close is usually not checked for return value, as there is little you can do if close fails. 

That said, the current code only tries to issue a warning if such happens. However... it won't work (if I am correct in reading the code). 

https://github.com/OpenLightingProject/ola/blob/master/plugins/uartdmx/UartWidget.cpp#L86

Close is documented to return -1 on error, and zero on success. That means > 0 will never happen. 

Maybe **!= 0** was intended. I personally like to check for **< 0**, allowing for calls to return error status-es other than -1, but for over 30 years now, most system functions are documented to return -1 on error and nothing else.... So my personal preference: 
`if (close(m_fd) < 0) {`
probably more consistent with other code: 
`if (close(m_fd) != 0) {`
otherwise valid: 
`if (close(m_fd) == -1) {`

**Comment by peternewman** [comment]
It looks like we've gone for a mix of:
`if (close(m_fd)) {`
and
```
int ret = close(m_fd);
if (ret == 0) {
```
and
```
int ret = close(m_fd);
if (ret != 0) {
```

In other code.

Do you want to open a Pull Request and apply the fix @rewolff ?

**Comment by peternewman** [comment]
@rewolff are you willing to put in the fix for this then?

**Comment by rewolff** [comment]
Not now. 
Keep reminding me. One of these days I'll get fed up and do it. Ok?
(I sell DMX boards and we've sold a batch that ALL don't work. Recall or try to fix on customer premises.... )

**Comment by peternewman** [comment]
Okay @rewolff . Thanks for thinking about it. :) We've got 0.10.3 coming out imminently if you fancied getting it into that. :p

Something major on the boards, or just a missing link?

**Comment by rewolff** [comment]
I'm going for 
`if (close (...) ) `
The reason being that it is the shortest. It is a common idiom, and if you have to learn that idiom, you need to learn it only once throughout the whole code if we apply it constently.
